lib/ash/test.ex

```diff
@@ -84,11 +95,43 @@ defmodule Ash.Test do
   Use the optional second argument to assert that the errors (all together) are of a specific class.
   """
   @spec refute_has_error(
-          Ash.Changeset.t() | Ash.Query.t() | Ash.ActionInput.t(),
+          Ash.Changeset.t() | Ash.Query.t() | Ash.ActionInput.t() | {:ok, term} | {:error, term},
           error_class :: Ash.Error.class_module(),
           (Ash.Error.t() -> boolean)
         ) :: Ash.Error.t() | no_return
-  def refute_has_error(changeset_query_or_input, error_class \\ nil, callback, opts \\ []) do
+  def refute_has_error(changeset_query_or_input, error_class \\ nil, callback, opts \\ [])
+
+  # An :ok tuple doesn't have any errors!
+  def refute_has_error({:ok, _record}, _error_class, _callback, _opts), do: :ok
+
+  def refute_has_error({:error, %{splode: splode} = error}, error_class, callback, opts) do
+    error = splode.to_class(error)
```

Comment:
We can safely assume these are Ash errors I think, so a shortcut here is `Ash.Error.to_error_class`